pkg/controller/restore_controller.go

```diff
@@ -462,43 +461,97 @@ func mostRecentCompletedBackup(backups []*api.Backup) *api.Backup {
 	return nil
 }
 
-func (c *restoreController) fetchBackup(objectStore cloudprovider.ObjectStore, name string) (*api.Backup, error) {
-	backup, err := c.backupLister.Backups(c.namespace).Get(name)
-	if err == nil {
-		return backup, nil
+// fetchBackupInfo checks the object storage for a backup that matches the given name. If it doesn't
+// find it, it tries to retrieve it from one of the backup storage locations.
+func (c *restoreController) fetchBackupInfo(backupName string, pluginManager plugin.Manager) (backupInfo, error) {
+	var info backupInfo
+	var err error
+	info.backup, err = c.backupLister.Backups(c.namespace).Get(backupName)
+	if err != nil {
+		return c.fetchFallBack(backupName, pluginManager)
 	}
 
-	if !apierrors.IsNotFound(err) {
-		return nil, errors.WithStack(err)
+	if info.backup.Spec.StorageLocation == "" {
```

Comment:
this block should be removed -- backups' storage locations are defaulted when the backup is created